File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
                int n = points.size(),c=0;
           sort(points.begin(), points.end(),
         [](const vector<int>& a, const vector<int>& b) {
             return a[1] < b[1];
         });
        int y=points[0][1];
        for(int i=1;i<n;i++)
        {
            if(y>=points[i][0])
            {
               
            }
            else{
                c++;
               
                y=points[i][1];
        
            }

        }
        c++;
        return c;
    
    }
};

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points,(a,b)->Integer.compare(a[0], b[0]));
        int n = points.length;
        int j=0;
        int ans=1;

        for (int i = 1; i < n; i++) {

            if (points[i][0] <= points[j][1]) {

                points[j][1] = Math.min(points[j][1], points[i][1]);

            } else {

                ans++;
                j = i;
            }
        }
        return ans;
    }
}